### PR TITLE
[SOAR-17800] Advanced Regex - Initial updates for fedramp compliance | Updated SDK to the latest version (#2839)

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Initial updates for fedramp compliance
  - Updated SDK to the latest version

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/advanced_regex/plugin.spec.yaml [hunk 1]
```diff
@@ -4,23 +4,43 @@ products: [insightconnect]
...
+  - Search and replace text
+  - Split text
+version_history:
+  - "1.0.4 - Initial updates for fedramp compliance | Updated SDK to the latest version"
+  - "1.0.3 - Update to make replace string non-required"
+  - "1.0.2 - Update to v4 Python plugin runtime | Add example inputs"
+  - "1.0.1 - New spec and help.md format for the Extension Library"
+  - "1.0.0 - Initial plugin"
+links:
+  - "[Python Regex](https://docs.python.org/library/re.html)"
```
Comment (lcwiklinski-r7) [inline plugins/advanced_regex/plugin.spec.yaml:41]: you can put the repository link like below:

`- "[Advanced Regex](https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/advanced_regex)"`

**Review by lcwiklinski-r7 — COMMENTED** [review]
It also looks like your hashes between provided plugin and checksum were not equal

**Comment by igorski-r7** [comment]
> It also looks like your hashes between provided plugin and checksum were not equal

It's because of non-generic type being held by the plugin response which is "[][]string". So I had to leave previously `schema.py` file to achieve backward compatibility. In that case I think it's ready to go.

File: plugins/advanced_regex/icon_advanced_regex/actions/replace/action.py [hunk 2]
```diff
@@ -16,9 +16,14 @@ def __init__(self):
...
-        flags = shared.constructFlags(params)
-        regex = params.get(Input.IN_REGEX)
-        new = params.get(Input.REPLACE_STRING, "")
-        replace_num = params.get(Input.MAX_REPLACE)
-        replaced = re.sub(regex, new, params.get(Input.IN_STRING), count=replace_num, flags=flags)
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        regex = params.get(Input.IN_REGEX, "")
+        input_string = params.get(Input.IN_STRING, "")
+        replace_string = params.get(Input.REPLACE_STRING, "")
+        replace_number = params.get(Input.MAX_REPLACE)
```
Comment (rmurray-r7) [inline plugins/advanced_regex/icon_advanced_regex/actions/replace/action.py:23]: Would it be best adding .strip()?

File: plugins/advanced_regex/icon_advanced_regex/actions/split/action.py [hunk 3]
```diff
@@ -16,8 +16,11 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        flags = shared.constructFlags(params)
-        regex = params.get(Input.IN_REGEX)
-        split = re.split(regex, params.get(Input.IN_STRING), maxsplit=params.get(Input.MAX_SPLIT), flags=flags)
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        regex = params.get(Input.IN_REGEX, "")
+        input_string = params.get(Input.IN_STRING, "")
```
Comment (rmurray-r7) [inline plugins/advanced_regex/icon_advanced_regex/actions/split/action.py:21]: same here 🤔

**Review by ekelly-r7 — APPROVED** [review]

File: plugins/advanced_regex/icon_advanced_regex/actions/replace/action.py (hunk 2 above)
Comment (igorski-r7) [reply plugins/advanced_regex/icon_advanced_regex/actions/replace/action.py:23]: I would leave it here as it is to be backward compatible (if any customer uses replace with space or something, same as for regex)

File: plugins/advanced_regex/plugin.spec.yaml (hunk 1 above)
Comment (igorski-r7) [reply plugins/advanced_regex/plugin.spec.yaml:41]: Imo we can skip it for now

**Pull request merged**